frontend/src/pages/NewRunV2.tsx

```diff
@@ -131,30 +143,50 @@ function NewRunV2(props: NewRunV2Props) {
   const [isStartingNewRun, setIsStartingNewRun] = useState(false);
   const [errorMessage, setErrorMessage] = useState('');
   const [isParameterValid, setIsParameterValid] = useState(false);
-  const [isRecurringRun, setIsRecurringRun] = useState(false);
-  const [trigger, setTrigger] = useState<ApiTrigger>();
-  const [maxConcurrentRuns, setMaxConcurrentRuns] = useState('10');
+  const [isClone, setIsClone] = useState(false);
+  const [isRecurringRun, setIsRecurringRun] = useState(apiRecurringRun !== undefined);
+  const [trigger, setTrigger] = useState(apiRecurringRun ? apiRecurringRun.trigger : undefined);
+  const [maxConcurrentRuns, setMaxConcurrentRuns] = useState(
+    apiRecurringRun ? apiRecurringRun.max_concurrency : '10',
+  );
   const [isMaxConcurrentRunValid, setIsMaxConcurrentRunValid] = useState(true);
-  const [catchup, setCatchup] = useState(true);
+  const initialCatchup =
+    apiRecurringRun && apiRecurringRun.no_catchup !== undefined
+      ? !apiRecurringRun.no_catchup
+      : true;
+  const [catchup, setCatchup] = useState(initialCatchup);
   const [clonedRuntimeConfig, setClonedRuntimeConfig] = useState<PipelineSpecRuntimeConfig>({});
 
   const urlParser = new URLParser(props);
-  const usePipelineFromRunLabel = 'Using pipeline from existing run.';
-  const pipelineDetailsUrl = existingRunId
+  const labelTextAdjective = isRecurringRun ? 'recurring ' : '';
+  const usePipelineFromRunLabel = `Using pipeline from existing ${labelTextAdjective} run.`;
+  const pipelineDetailsUrlfromRun = existingRunId
     ? RoutePage.PIPELINE_DETAILS.replace(
         ':' + RouteParams.pipelineId + '/version/:' + RouteParams.pipelineVersionId + '?',
         '',
       ) + urlParser.build({ [QUERY_PARAMS.fromRunId]: existingRunId })
     : '';
 
+  const pipelineDetailsUrlfromRecurringRun = originalRecurringRunId
+    ? RoutePage.PIPELINE_DETAILS.replace(
+        ':' + RouteParams.pipelineId + '/version/:' + RouteParams.pipelineVersionId + '?',
+        '',
+      ) + urlParser.build({ [QUERY_PARAMS.cloneFromRecurringRun]: originalRecurringRunId })
+    : '';
+
+  const pipelineDetailsUrl = pipelineDetailsUrlfromRun
+    ? pipelineDetailsUrlfromRun
+    : pipelineDetailsUrlfromRecurringRun;
   const isTemplatePullSuccess = templateString ? true : false;
-  const apiResourceRefFromRun = apiRun?.run?.resource_references
-    ? apiRun.run?.resource_references
-    : undefined;
+  const apiResourceRefFromRun = apiRun?.run?.resource_references;
+  const apiResourceRefFromRecurringRun = apiRecurringRun?.resource_references;
+  const existResourceRef = apiResourceRefFromRun
```

Comment:
Looks like `apiResourceRefFromRun` takes higher priority than `apiResourceRefFromRecurringRun`, while it is not the case in `NewRunSwitcher` file. Let's be clear about which one takes higher priority, or even better, makes sure that `apiRun` and `apiRecurringRun` don't exist at the same time.